api/app/signals/apps/signals/v1/private/views.py

```diff
@@ -85,46 +58,23 @@ class PrivateSignalSplitViewSet(mixins.CreateModelMixin, mixins.RetrieveModelMix
     queryset = Signal.objects.all()
 
 
-class PublicSignalViewSet(mixins.CreateModelMixin,
-                          DetailSerializerMixin,
-                          mixins.RetrieveModelMixin,
-                          viewsets.GenericViewSet):
-    queryset = Signal.objects.all()
-    serializer_class = PublicSignalCreateSerializer
-    serializer_detail_class = PublicSignalSerializerDetail
-    lookup_field = 'signal_id'
-
-
-class PublicSignalAttachmentsViewSet(mixins.CreateModelMixin, viewsets.GenericViewSet):
+class PrivateSignalAttachmentsViewSet(mixins.CreateModelMixin, mixins.ListModelMixin,
+                                      viewsets.GenericViewSet):
     serializer_class = SignalAttachmentSerializer
-    signal = None
-    is_public = True
-    lookup_field = 'signal_id'
-    lookup_url_kwarg = 'signal_id'
-
-    def _get_signal(self):
-        if self.signal is None:
-            self.signal = Signal.objects.get(
-                **{self.lookup_field: self.kwargs[self.lookup_url_kwarg]})
-
-        return self.signal
-
-    def get_serializer_context(self):
-        context = super().get_serializer_context()
-        context['signal'] = self._get_signal()
-        context['is_public'] = self.is_public
+    pagination_class = None
 
-        return context
+    _signal = None
 
+    @property
+    def signal(self):
+        if not self._signal:
```

Comment:
move to `get_object` method (see comment below)

Reply:
Done